lib/chewy/index/crutch.rb

```diff
@@ -12,14 +12,15 @@ class Crutches
         def initialize(index, collection)
           @index = index
           @collection = collection
+          @storage = {}
           @index._crutches.each_key do |name|
-            singleton_class.class_eval <<-METHOD, __FILE__, __LINE__ + 1
-              def #{name}
-                @#{name} ||= @index._crutches[:#{name}].call @collection
-              end
-            METHOD
+            @storage[name] = @index._crutches[:"#{name}"].call @collection
           end
         end
+
+        def fetch(name)
+          @storage[name]
+        end
```

Comment:
In that case we don't break the syntax.

```suggestion
        end

        def method_missing(name, *, **)
          return self[name] if @index._crutches.key?(name)

          super
        end
        
        def respond_to_missing?(name, include_private = false)
          @index._crutches.key?(name) || super
        end

        def [](name)
          @crutches[name] ||= @index._crutches[:"#{name}"].call(@collection)
        end
```

Reply:
Thanks, yeah, looks better and we do lazy instantiation as before. Although in our code base I prefer to use `[]` hash access method, since it's called literally on each field and method_missing gives a performance penalty.